pymc/distributions/mixture.py

```diff
@@ -367,6 +373,25 @@ def marginal_mixture_logprob(op, values, rng, weights, *components, **kwargs):
     return mix_logp
 
 
+@_get_moment.register(MarginalMixtureRV)
+def get_moment_marginal_mixture(op, rv, rng, weights, *components):
+    if len(components) == 1:
+        ndim_supp = components[0].owner.op.ndim_supp
+
+        # assuming weights.shape[-1] is the mixing component
+        # trying to expand dimensions by ndim_supp inside out
+        expansion_axes = [-i for i in range(ndim_supp, 0, -1)]
+
+        moment_components = get_moment(components[0])
+
+        weights = at.expand_dims(weights, axis=expansion_axes)
+
+    else:
+        moment_components = at.stack([get_moment(component) for component in components], axis=-1)
+
+    return at.sum(weights * moment_components, axis=-1)
```

Comment:
I think this sum should be done across the mixture axis, which is not necessarily the last one for single component mixtures

Reply:
Would it be `axis = -ndim_supp - 1`?

Reply:
Yeah, if you stack the `moment_components` along that axis as well